File: frontend/src/app/foundation/mainstack/kumojin-events/models/kumojin-event.ts
import { autoserialize, autoserializeAs, deserialize } from "cerialize";

export class KumojinEvent {
  @deserialize public guid = '';

  @autoserialize public displayName = '';
  @autoserialize public description = '';
  @autoserialize public countryAt = '';

  @autoserializeAs(Date) public startAt?: Date;
  @autoserializeAs(Date) endAt?: Date;
}

File: backend/Kumojin.Backend.Domain/KumojinEvents/KumojinEvent.cs
using System;
using System.ComponentModel.DataAnnotations;
using Kumojin.Backend.Domain.Bases;

namespace Kumojin.Backend.Domain.KumojinEvents
{
    public class KumojinEvent : BaseModel
    {
        [MaxLength(32)]
        public string DisplayName { get; set; }
        public string Description { get; set; }

        public DateTime StartAt { get; set; }
        public DateTime EndAt { get; set; }

        public string CountryAt { get; set; }
    }
}